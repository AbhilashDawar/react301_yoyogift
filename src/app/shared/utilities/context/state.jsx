import * as React from 'react';

let ContextState = React.createContext();

let initialState = {
    showLoader: false,
    // filterParams: {
    //     startValue: config.rangeDefaults.min,
    //     endValue: config.rangeDefaults.max,
    //     discountOption: 0
    // }
};

let reducer = (state, action) => {
    switch (action.type) {
        case "RESET":
            return initialState;
        case "SHOW":
            return { ...state, showLoader: true };
        case "HIDE":
            return { ...state, showLoader: false };
        // case "UPDATE_FILTERS":
        //     return { ...state, filterParams: action.payload };
        default:
            return { ...state };
    }
};

function ContextStateProvider(props) {
    let [state, dispatch] = React.useReducer(reducer, initialState);
    let value = { state, dispatch };

    return (
        <ContextState.Provider value={value}>{props.children}</ContextState.Provider>
    );
}

let ContextStateConsumer = ContextState.Consumer;

export { ContextState, ContextStateProvider, ContextStateConsumer };