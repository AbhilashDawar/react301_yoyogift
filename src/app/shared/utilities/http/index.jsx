import axios from 'axios';

const GET = (endpoint) => {
    return axios.get(endpoint);
}

const POST = (endpoint, payload) => {
    return axios.post(endpoint, payload);
}

const PUT = (endpoint, payload) => {
    return axios.put(endpoint, payload);
}

const PATCH = (endpoint, payload) => {
    return axios.patch(endpoint, payload);
}

const DELETE = (endpoint) => {
    return axios.delete(endpoint);
}

export default {
    GET,
    POST,
    PUT,
    PATCH,
    DELETE
}
