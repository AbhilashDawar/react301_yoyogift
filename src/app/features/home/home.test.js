import React from 'react';
import { shallow, mount } from 'enzyme';
import { Typography } from '@material-ui/core';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { ContextStateProvider } from '../../shared/utilities/context/state.jsx';
import Home from './index.jsx';
import http from '../../shared/utilities/http';

jest.mock('axios');
describe('<Home />', () => {

    it('should have one Typography element', () => {
        jest.spyOn(http, 'GET').mockImplementation(() => Promise.resolve([
            {
                "id": "17-909-7507",
                "name": "DOYLE-NITZSCHE",
                "brand": "Levi's",
                "desc": "Quisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.\n\nPhasellus in felis. Donec semper sapien a libero. Nam dui.",
                "imageUrl": "https://shop.r10s.jp/marukawa7/cabinet/test86/0117000298-20.jpg",
                "buyoutPoints": 1000,
                "categoryId": 1,
                "rating": 1,
                "discounts": 45
            },
            {
                "id": "17-909-7507",
                "name": "DOYLE-NITZSCHE",
                "brand": "Levi's",
                "desc": "Quisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.\n\nPhasellus in felis. Donec semper sapien a libero. Nam dui.",
                "imageUrl": "https://shop.r10s.jp/marukawa7/cabinet/test86/0117000298-20.jpg",
                "buyoutPoints": 1000,
                "categoryId": 1,
                "rating": 1,
                "discounts": 45
            },
            {
                "id": "17-909-7507",
                "name": "DOYLE-NITZSCHE",
                "brand": "Levi's",
                "desc": "Quisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.\n\nPhasellus in felis. Donec semper sapien a libero. Nam dui.",
                "imageUrl": "https://shop.r10s.jp/marukawa7/cabinet/test86/0117000298-20.jpg",
                "buyoutPoints": 1000,
                "categoryId": 1,
                "rating": 1,
                "discounts": 45
            },
            {
                "id": "17-909-7507",
                "name": "DOYLE-NITZSCHE",
                "brand": "Levi's",
                "desc": "Quisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.\n\nPhasellus in felis. Donec semper sapien a libero. Nam dui.",
                "imageUrl": "https://shop.r10s.jp/marukawa7/cabinet/test86/0117000298-20.jpg",
                "buyoutPoints": 1000,
                "categoryId": 1,
                "rating": 1,
                "discounts": 45
            },
            {
                "id": "17-909-7507",
                "name": "DOYLE-NITZSCHE",
                "brand": "Levi's",
                "desc": "Quisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.\n\nPhasellus in felis. Donec semper sapien a libero. Nam dui.",
                "imageUrl": "https://shop.r10s.jp/marukawa7/cabinet/test86/0117000298-20.jpg",
                "buyoutPoints": 1000,
                "categoryId": 1,
                "rating": 1,
                "discounts": 45
            },
            {
                "id": "17-909-7507",
                "name": "DOYLE-NITZSCHE",
                "brand": "Levi's",
                "desc": "Quisque porta volutpat erat. Quisque erat eros, viverra eget, congue eget, semper rutrum, nulla. Nunc purus.\n\nPhasellus in felis. Donec semper sapien a libero. Nam dui.",
                "imageUrl": "https://shop.r10s.jp/marukawa7/cabinet/test86/0117000298-20.jpg",
                "buyoutPoints": 1000,
                "categoryId": 1,
                "rating": 1,
                "discounts": 45
            },
        ]));
        act(() => {
            const component = mount(
                <MemoryRouter>
                    <ContextStateProvider>
                        <Home />
                    </ContextStateProvider>
                </MemoryRouter>
            );
        })
        expect(component.find(Typography)).toHaveLength(1);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <ContextStateProvider>
                    <Home />
                </ContextStateProvider>
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
