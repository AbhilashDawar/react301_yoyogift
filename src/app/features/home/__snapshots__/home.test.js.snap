// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-div-2 MuiPaper-rounded"
/>
`;
